### PR TITLE
CL-2 Gather all necessary cryptocurrency network information

### PR DESCRIPTION
**Pull request by szerepak**
Added cryptocurrency networks:
1. BCH (bitcoin cash)
2. BTC (bitcoin core)
3. DASH (dashcoin)
4. DOGE (dogecoin)
5. KMD (komodo)
6. LTC (litecoin)
7. MONA (monacoin)
8. ZCL (zclassic) *official dns seeds cannot be resolved*

### CODE REVIEW

File: clove/network/base.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+class BaseNetwork(object):
+    name = None
+    symbols = ()
+    seeds = ()
+    port = None
+
+    @property
+    def default_symbol(self):
+        if len(self.symbols) > 0:
```
Comment (trojkat) [inline clove/network/base.py]: `if self.symbols:` should be enough

File: clove/network/base.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    def get_nodes(cls, index=0) -> list:
+        try:
+            hostname, alias, nodes = socket.gethostbyname_ex(cls.seeds[index])
+        except (socket.herror, socket.gaierror):
+            return []
+
+        return nodes
+
+    @classmethod
+    def connect(cls, timeout):
```
Comment (trojkat) [inline clove/network/base.py]: Let's set some default timeout - 2s.

File: clove/network/base.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        index = 0
+        while index < len(cls.seeds):
+            nodes = cls.get_nodes(index)
+
+            for node in nodes:
+                try:
+                    connection = socket.create_connection(
```
Comment (trojkat) [inline clove/network/base.py]: We should check if it will work for ipv6.

File: clove/network/zclassic.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
+from clove.network.base import BaseNetwork
+
+
+class ZClassic(BaseNetwork):
```
Comment (trojkat) [inline clove/network/zclassic.py]: `Zclassic`

File: clove/network/base.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+            hostname, alias, nodes = socket.gethostbyname_ex(cls.seeds[index])
+        except (socket.herror, socket.gaierror):
+            return []
+
+        return nodes
+
+    @classmethod
+    def connect(cls, timeout=2):
+        index = 0
+        while index < len(cls.seeds):
```
Comment (pafff) [inline clove/network/base.py]: Maybe just `for seed in cls.seeds`?

File: clove/network/bitcoin_cash.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    """
+    Class with all the necessary BCH testing network information based on
+    https://github.com/Bitcoin-ABC/bitcoin-abc/blob/master/src/chainparams.cpp
+    (date of access: 01/18/2018)
+    """
+    name = 'test-bitcoin-cash'
+    seeds = (
+        'testnet-seed.bitcoinabc.org',
+        'testnet-seed-abc.bitcoinforks.org',
+        'testnet - seed.bitprim.org',
```
Comment (trojkat) [inline clove/network/bitcoin_cash.py:33]: We can create a test to check if every seed is a valid address.

**Pull request merged**